modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/AdaptiveScriptInitializerTestCase.java

```diff
@@ -0,0 +1,106 @@
+/*
+ *  Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.identity.integration.test.auth;
+
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.wso2.carbon.automation.extensions.servers.utils.ServerLogReader;
+import org.wso2.carbon.integration.common.utils.exceptions.AutomationUtilException;
+import org.wso2.carbon.integration.common.utils.mgt.ServerConfigurationManager;
+
+import javax.xml.xpath.XPathExpressionException;
+import java.io.File;
+import java.io.IOException;
+
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Test for temporary claim persistence with adaptive authentication scripts.
+ */
+public class AdaptiveScriptInitializerTestCase extends AbstractAdaptiveAuthenticationTestCase {
+
+    private ServerConfigurationManager serverConfigurationManager;
+
+    @BeforeTest(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init();
+        runAdaptiveAuthenticationDependencyScript();
+    }
+
+    private void runAdaptiveAuthenticationDependencyScript() {
+
+        ServerLogReader inputStreamHandler;
+        ServerLogReader errorStreamHandler;
+        String scriptFolder = (System.getProperty("carbon.home"))
+                + File.separator + "bin" + File.separator;
+        Process tempProcess = null;
+        File shFile = new File(scriptFolder);
+        Runtime rt = Runtime.getRuntime();
+
+        try {
+            if (System.getProperty("os.name").toLowerCase().contains("windows")) {
+                log.warn("Operating System is Windows, skipping execution");
+            } else {
+                log.info("Operating system is not windows. Executing shell script");
+                tempProcess = rt.getRuntime().exec(
+                        new String[] { "/bin/bash", "dependencydownloader.sh" }, null, shFile);
```

Comment:
better to come up with a different name? it is not simply a dependency downloader right ? are we planning to extend this to download new dependencies?